### PR TITLE
chore: add support for SESv2 Sigv4a (not yet used because SESv2 hasn't rolled out support)

### PR DESCRIPTION
**Pull request by ianbotsf**
## Issue \#

(none)

## Description of changes

This change prepares for SESv2 Sigv4a support. This won't take effect until additional support rolls out from the service side.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: services/sesv2/e2eTest/src/Sigv4aTest.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import kotlin.test.Test
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+
+class Sigv4aTest {
+    @Test
+    @Ignore // TODO enable once SESv2 model adds endpointId and Sigv4a
+
```
Comment (lauzadis) [inline services/sesv2/e2eTest/src/Sigv4aTest.kt:25]: nit: remove blank line

File: services/sesv2/e2eTest/src/Sigv4aTest.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                maxAttempts = 1 // The call is intended to fail, no sense trying more than once
+            }
+
+            interceptors += interceptor
+
+            authSchemes = listOf(SigV4AsymmetricAuthScheme(CrtAwsSigner, "ses"))
+        }.use { ses ->
+            assertFailsWith<HttpException> {
+                ses.sendEmail {
+                    // endpointId = "bdm3x3zl.n5x" // TODO uncomment
```
Comment (lauzadis) [inline services/sesv2/e2eTest/src/Sigv4aTest.kt:40]: Will this TODO be completed in this PR?

File: services/sesv2/e2eTest/src/Sigv4aTest.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                }
+            }
+        }
+
+        assertEquals(1, interceptor.requests.size)
+        val request = interceptor.requests.single()
+
+        assertContains("bdm3x3zl.n5x.endpoints.email.amazonaws.com", request.url.host.toString()) // Correct endpoint?
```
Comment (lauzadis) [inline services/sesv2/e2eTest/src/Sigv4aTest.kt:48]: Is it the correct endpoint?

**Review by lauzadis — APPROVED** [review]

File: services/sesv2/e2eTest/src/Sigv4aTest.kt (hunk 2 above)
Comment (ianbotsf) [reply services/sesv2/e2eTest/src/Sigv4aTest.kt:40]: No, we'll need to follow up. There's no public issue to track for doing so.

File: services/sesv2/e2eTest/src/Sigv4aTest.kt (hunk 3 above)
Comment (ianbotsf) [reply services/sesv2/e2eTest/src/Sigv4aTest.kt:48]: Yes, it is. The comment is probably misleading now, I'll update it.

**Review by 0marperez — APPROVED** [review]

**Pull request merged**